pkg/dns/serviceresolver.go

```diff
@@ -210,14 +207,6 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 			return nil, errNoSuchName
 		}
 
-		hostnameMappings := noHostnameMappings
```

Comment:
did upstream drop support for the annotations entirely? are we matching them here? @smarterclayton PTAL